h2o-py/h2o/explain/_explain.py

```diff
@@ -0,0 +1,1763 @@
+# -*- encoding: utf-8 -*-
+
+import random
+from contextlib import contextmanager
+from collections import OrderedDict, Counter
+
+import h2o
+import numpy as np
+import matplotlib
+import matplotlib.colors
+import matplotlib.figure
+try:
+    import matplotlib.pyplot as plt
+except ImportError:
+    # Possibly failed due to missing tkinter in old matplotlib in python 2.7
+    matplotlib.use("Agg")
+    import matplotlib.pyplot as plt
+
+
+def _display(object):
+    """
+    Display the object.
+    :param object: An object to be displayed.
+    :return: the input
+    """
+    try:
+        import IPython.display
+        IPython.display.display(object)
+    except ImportError:
+        if isinstance(object, matplotlib.figure.Figure):
+            plt.show()
+        else:
+            print(object)
+    if isinstance(object, matplotlib.figure.Figure):
+        plt.close(object)
+    return object
+
+
+def _dont_display(object):
+    """
+    Don't display the object
+    :param object: that should not be displayed
+    :return: input
+    """
+    if isinstance(object, matplotlib.figure.Figure):
+        plt.close()
+    return object
+
+
+# UTILS
+class Header:
+    """
+    Class representing a Header with pretty printing for IPython.
+    """
+
+    def __init__(self, content, level=1):
+        self.content = content
+        self.level = level
+
+    def _repr_html_(self):
+        return "<h{level}>{content}</h{level}>".format(level=self.level, content=self.content)
+
+    def _repr_markdown_(self):
+        return "\n\n{} {}".format("#" * self.level, self.content)
+
+    def __repr__(self):
```

Comment:
usually, better implement `__str__` for printing and keep `__repr__` for a parseable representation.

Reply:
Good point! I must have confused myself with the IPython's _repr methods.